File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>>h_map;
        vector<vector<string>>answer;
        for(string str: strs){
            string temp;
            temp = str;
            sort(temp.begin(), temp.end());
            h_map[temp].push_back(str);
        }

        for(auto& [sorted,original]: h_map){
            answer.push_back(original);
        }
        return answer;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        hashMap = {}

        for string in strs:
            sortedstr = ''.join(sorted(string))
            if sortedstr not in hashMap:
                hashMap[sortedstr] = []
            hashMap[sortedstr].append(string)
        ans = []
        for key, val in hashMap.items():
            ans.append(val)
        return ans
